ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderBinomialTestSuite.scala

```diff
@@ -118,6 +126,8 @@ class H2OTargetEncoderBinomialTestSuite extends FunSuite with Matchers with Shar
       .setHoldoutStrategy("None")
       .setBlendedAvgEnabled(true)
       .setNoise(0.0)
+      .setProblemType("Binomial")
```

Comment:
is setting problem type mandatory? if it is - can we auto-detect it and allow old code to run?

Reply:
It's not mandatory. Problem typy is [binomial by default](https://github.com/h2oai/sparkling-water/pull/2462/files/aa72f6e665bd80005d79d9836e293a16ba68e607#diff-de07ee0fa2b229cdb67aee97213ff7d1e27a9089c556e9ec92b02c84df6c29a3R91). For automatic detection, I would need to change the output type for binomial and regression to a vector since [here](https://github.com/h2oai/sparkling-water/pull/2462/files#diff-5c5fcf2c0792550b3e9863f7c39fbcf432ef350bef92b70f4793830cc1fe7e4aR26-R33) I don't have an access to a dataset. This might be a breaking change to some level, but people using TE in pipeline with a H2O algorithm or [VectorAssembler](https://spark.apache.org/docs/latest/ml-features#vectorassembler) (Spark algorithms take vector as an input) wouldn't see any difference IMHO.